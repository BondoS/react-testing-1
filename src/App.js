import React, {Component, setState} from 'react';
import logo from './logo.svg';
import './App.css';

const Test = () => <div>Testing</div>;

class App extends Component {
  state = {
    on: false,
    input: '',
    mainColor: 'blue'
  };
  render () {
    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          <p>
            Edit <code>src/App.js</code> and save to reload.
          </p>
          <a
            className="App-link"
            href="https://reactjs.org"
            target="_blank"
            rel="noopener noreferrer"
          >
            Learn React
          </a>
        </header>
        <h3 className={this.state.mainColor}>Everyone is Welcome!</h3>
        <p className="button-state">{this.state.on ? 'Yes!' : 'No!'}</p>
        <button
          onClick={() =>
            this.setState (prev => {
              return {
                on: !prev.on,
              };
            })}
        >
          Click
        </button>
        <h2>{this.state.input}</h2>
        <input
          type="text"
          onChange={e => this.setState ({input: e.currentTarget.value})}
        />
        <Test />
      </div>
    );
  }
}

export class Link extends Component {
  render () {
    return this.props.hide ? null : <a href={this.props.address}>Click</a>;
  }
}

export default App;
