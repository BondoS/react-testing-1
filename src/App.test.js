import React from 'react';
import ReactDOM from 'react-dom';
import App, {Link} from './App';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import expectExport from 'expect';
import toJson from 'enzyme-to-json';

configure ({adapter: new Adapter ()});

describe ('<App />', () => {
  it ('should contain correct number of elements', () => {
    const wrapper = shallow (<App />);
    expectExport (wrapper.find ('.App-header').children ().length).toBe (3);
  });
  it ('matches the snapshot', () => {
    const tree = shallow (<App />);
    expect (toJson (tree)).toMatchSnapshot ();
  });
});

describe ('<Link />', () => {
  it ('link component accepts address prop', () => {
    const wrapper = shallow (<Link address="www.google.com" />);
    expect (wrapper.instance ().props.address).toBe ('www.google.com');
  });
  it ('a tag renders href correctly', () => {
    const wrapper = shallow (<Link address="www.google.com" />);
    expect (wrapper.props ().href).toBe ('www.google.com');
  });
  it ('returns null with true hide prop', () => {
    const wrapper = shallow (<Link hide={false} />);
    expect (wrapper.find ('a').length).toBe (1);
    wrapper.setProps ({hide: true});
    expect (wrapper.get (0)).toBeNull ();
  });
});
