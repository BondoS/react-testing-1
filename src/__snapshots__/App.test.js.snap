// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> mount rendering matches the snapshot 1`] = `
<App>
  <div
    className="App"
  >
    <header
      className="App-header"
    >
      <img
        alt="logo"
        className="App-logo"
        src="logo.svg"
      />
      <p>
        Edit 
        <code>
          src/App.js
        </code>
         and save to reload.
      </p>
      <a
        className="App-link"
        href="https://reactjs.org"
        rel="noopener noreferrer"
        target="_blank"
      >
        Learn React
      </a>
    </header>
    <Form>
      <form
        data-testid="addUserForm"
        onSubmit={[Function]}
      >
        <h2>
          Request Information
        </h2>
        <input
          data-testid="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          data-testid="email"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <input
          data-testid="number"
          onChange={[Function]}
          placeholder="Number"
          type="text"
          value=""
        />
        <div>
          <input
            checked={true}
            data-testid="checked"
            onChange={[Function]}
            onClick={[Function]}
            type="checkbox"
          />
          <p
            className="promotions"
            data-testid="promotionsP"
          >
            Receive Promotions
          </p>
        </div>
        <button
          data-testid="submitButton"
          type="submit"
        >
          Submit
        </button>
      </form>
    </Form>
    <h3
      className="blue"
    >
      Everyone is Welcome!
    </h3>
    <p
      className="button-state"
    >
      No!
    </p>
    <button
      onClick={[Function]}
    >
      Click
    </button>
    <h2 />
    <input
      onChange={[Function]}
      type="text"
    />
    <div>
      <a
        className="lifeCycle"
        href="/"
      >
        componentDidMount
      </a>
    </div>
    <Test>
      <div>
        Testing
      </div>
    </Test>
  </div>
</App>
`;

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <p>
      Edit 
      <code>
        src/App.js
      </code>
       and save to reload.
    </p>
    <a
      className="App-link"
      href="https://reactjs.org"
      rel="noopener noreferrer"
      target="_blank"
    >
      Learn React
    </a>
  </header>
  <Form />
  <h3
    className="blue"
  >
    Everyone is Welcome!
  </h3>
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
    type="text"
  />
  <div>
    <a
      className="lifeCycle"
      href="/"
    >
      componentDidMount
    </a>
  </div>
  <Test />
</div>
`;
